### PR TITLE
Add script post publish to update bpk deps in the docs site

### PR DESCRIPTION
**Pull request by olliecurtis**
So that we don't need to worry about the docs site becoming out of date with Web components or updates (e.g. icons being released), created a simple script that will run post publishing our packages so that we don't forget 😄 

The script is **not** made to be super intense and do loads of bells and whilstes but just simply take the effort out of changing projects, updating and creating a PR through the website.

The script will: 

- clone the docs site locally, 
- run an update for bpk packages, 
- install the new packages 
- use the gh cli to push and create an automatic PR to the docs site
- Cleanup after itself (litterally just an `rm -rf`)

To show it working a sample PR is available here: https://github.com/olliecurtis/docs-testground/pull/4

### CODE REVIEW

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [inline scripts/publish-process/update-docs-deps.sh:27]: This will fail if the branch already exists remotely. Maybe it should be this + timestamp?

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [inline scripts/publish-process/update-docs-deps.sh:38]: This assumes the publisher is signed in to GH for command line. Probably an acceptable assumption.

Do we need to set the title? By default doesn't it use the commit message?

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:27]: Only the case if you doon't delete the branch after merging the PRs ;)

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [inline scripts/publish-process/update-docs-deps.sh:30]: I feel like maybe the backpack-docs repo should have it's own script for bumping which this script could just use? But maybe that's over engineering it?

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:38]: Yeah I guess we can exclude the title - I don't think personally it matters what we do for this and if we need to get picky about setting it or not

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:30]: Yeah - I don't think we need to over engineer here, this is just a simple script to help out, as the docs site would never auto run a script.

I just translated my workflow into a script here for convience and having steps across both repos adds to the maintenace burden which we want to avoid

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [reply scripts/publish-process/update-docs-deps.sh:27]: Which could easily happen, especially as we sometimes release a lot

Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:27]: Its also worth noting we do have this setting turned on in the docs repo meaning that the branch should never exist on remote once the PR has been merged 🤷 

Screenshot 2021-03-25 at 13 38 27

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:30]: I do also however see we have the npx part as a npm command we can invoke so could just use that instead

https://github.com/Skyscanner/backpack-docs/blob/main/package.json#L45

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [reply scripts/publish-process/update-docs-deps.sh:27]: Yeah but suppose I release something small, I might not look at the docs at all, and then someone releases something later and suddenly the PR they actually want isn't made because the branch already existed

File: scripts/publish-process/update-docs-deps.sh
Comment (georgegillams) [reply scripts/publish-process/update-docs-deps.sh:30]: Yeah. I guess maybe that should do the `npm i` as well as it hasn't technically upgraded them until that's done

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:27]: That is true, I just went on the assumption our PRs channel would alert that a PR has been opened and we would be jumping on it asap as normal, but I do see this as a potential clash!

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:30]: Docs site script has been fixed to include this and now the step has been simplified! Thanks for the suggestion looks cleaner now!

File: scripts/publish-process/update-docs-deps.sh
Comment (olliecurtis) [reply scripts/publish-process/update-docs-deps.sh:27]: Changed this to use the date and time at the end to avoid the problem @georgegillams has mentioned above

**Review by georgegillams — APPROVED** [review]
Excellent

**Review by shaundon — CHANGES_REQUESTED** [review]
I have a couple of small suggestions so I'm just sending this now to block the PR before it automerges 😅 I'll add full comment in a moment

File: .gitignore
Comment (shaundon) [inline .gitignore:20]: Good to have this comment here, but I could foresee a situation in the future where somebody new doesn't know what we mean by 'the script', so I've expanded the comment to add some more context.

```suggestion
# On postpublish, we clone and run some scripts on the backpack-docs repo. This ensures we don't accidentally commit it.
```

File: package.json
Comment (shaundon) [inline package.json:52]: npm has a built-in `postpublish` script that runs automatically after `publish` completes, could we use that for this instead? It'll just help the command from getting too long, and make more sense semantically.

```suggestion
    "publish": "npm run check-pristine && npm run check-owners && npm run build && git pull --rebase && flow stop && npm run test && lerna publish && npm run publish:css",
    "postpublish": "./scripts/publish/process/update-docs-deps.sh",
```

File: scripts/publish-process/update-docs-deps.sh
Comment (shaundon) [inline scripts/publish-process/update-docs-deps.sh:22]: `tmp` discussion.

File: package.json
Comment (olliecurtis) [reply package.json:52]: Let me take a look at this and see what happens - now I saw it in the npm docs will give it a go!

**Review by shaundon — APPROVED** [review]

**Pull request merged**
